sormas-api/src/main/java/de/symeda/sormas/api/caze/CaseLogic.java

```diff
@@ -53,15 +55,21 @@ public static void validateInvestigationDoneAllowed(CaseDataDto caze) throws Val
 		}
 	}
 
-	public static Date getStartDate(Date onsetDate, Date reportDate) {
+	public static FollowUpPeriodDto getStartDate(CaseDataDto caseDto, List<SampleDto> samples) {
+		return getStartDate(caseDto.getSymptoms().getOnsetDate(), caseDto.getReportDate(), samples);
+	}
+
+	public static FollowUpPeriodDto getStartDate(Date onsetDate, Date reportDate, List<SampleDto> samples) {
 
 		if (onsetDate != null) {
-			return onsetDate;
-		} else {
-			return reportDate;
+			return new FollowUpPeriodDto(onsetDate, FollowUpStartDateType.SYMPTOM_ONSET_DATE);
 		}
+		return FollowUpLogic.getStartDate(reportDate, samples);
 	}
 
+	// TODO: this is inconsistent with the result of calculateFollowUpUntilDate
```

Comment:
That's okay. `getEndDate` is used to retrieve the date after which the case is not considered "relevant" for follow-up related things anymore, so primarily the follow-up until date if available. Onset/report date are just fallbacks. Visits or samples aren't important for this calculation.